engine/access/rpc/backend/backend_network.go

```diff
@@ -80,6 +83,75 @@ func (b *backendNetwork) GetLatestProtocolStateSnapshot(_ context.Context) ([]by
 	return data, nil
 }
 
+// GetProtocolStateSnapshotByBlockID returns serializable Snapshot by blockID
+func (b *backendNetwork) GetProtocolStateSnapshotByBlockID(_ context.Context, blockID flow.Identifier) ([]byte, error) {
+	snapshotHeadByBlockId, err := b.state.AtBlockID(blockID).Head()
+	if err != nil {
+		if errors.Is(err, state.ErrUnknownSnapshotReference) {
+			return nil, status.Errorf(codes.NotFound, "failed to get a valid snapshot: block not found")
+		}
+
+		return nil, status.Errorf(codes.Internal, "failed to get a valid snapshot: block not found")
```

Comment:
provide some more details to the user about why this failed
```suggestion
		return nil, status.Errorf(codes.Internal, "failed to get block: %v", err)
```